internal/conditions/validate.go

```diff
@@ -0,0 +1,70 @@
+package conditions
+
+import (
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+	"github.com/giantswarm/azure-admission-controller/internal/errors"
+)
+
+type conditionChangeValidation func(oldCondition, newCondition *capi.Condition) error
+
+func validateAll(oldCondition, newCondition *capi.Condition, validations []conditionChangeValidation) error {
+	var err error
+
+	for _, validation := range validations {
+		err = validation(oldCondition, newCondition)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	return nil
+}
+
+func validateNewStatusUnknownIsNotAllowed(_, newCondition *capi.Condition) error {
```

Comment:
isn't this validation redundant because `validateNewStatusMustBeEitherTrueOrFalse` already validates is either true or false?

Reply:
It does, but I added a separate validation for `Unknown` on purpose, as `Unknown` status is defined in k8s core package and could be set more easily and on purpose, while we don't want that.

Other possible values, other than `Unknown`, `True` and `False`, could be some new status values in some new upstream package versions that we did not take into account yet, or simply a mistake somewhere in code (which could be also customer's code now with CP API access).

Reply:
ok, it makes sense :+1:

Reply:
So basically distinguishing between `Unknown` that we know about and other unknown status values :)